app/src/main/java/com/hedvig/app/feature/dismissiblepager/DismissiblePager.kt

```diff
@@ -59,57 +68,103 @@ abstract class DismissiblePager : DialogFragment() {
+            root.setEdgeToEdgeSystemUiFlags(true)
 
-        titleLabel?.let { tl ->
-            title.text = resources.getString(tl)
-            title.show()
-        }
+            proceed.doOnApplyWindowInsets { view, insets, initialState ->
+                view.updateMargin(bottom = initialState.margins.bottom + insets.systemWindowInsetBottom)
+            }
 
-        pager.adapter = DismissablePagerAdapter(childFragmentManager, items, requireContext())
-        pagerIndicator.pager = pager
-        proceed.text = if (items.size > 1) {
-            resources.getString(proceedLabel)
-        } else {
-            resources.getString(dismissLabel)
-        }
-        pager.addOnPageChangeListener(PageChangeListener())
-        proceed.setHapticClickListener {
-            tracker.clickProceed()
-            pager.currentItem += 1
+            topBar.doOnApplyWindowInsets { view, insets, initialState ->
+                view.updatePadding(top = initialState.paddings.top + insets.stableInsetTop)
+            }
+
+            pagerIndicator.shouldShowLogo = this@DismissiblePager.shouldShowLogo
+
+            close.setOnClickListener {
+                onDismiss()
+                dialog?.dismiss()
+            }
+
+            titleLabel?.let { tl ->
+                title.text = resources.getString(tl)
+                title.show()
+            }
+
+            pager.adapter = DismissiblePagerAdapter(childFragmentManager, items, requireContext())
+            pagerIndicator.pager = pager
+            proceed.text = if (items.size > 1) {
+                resources.getString(proceedLabel)
+            } else {
+                resources.getString(lastButtonText)
+            }
+            pager.addOnPageChangeListener(PageChangeListener())
+            proceed.setHapticClickListener {
+                tracker.clickProceed()
+                pager.currentItem += 1
+            }
         }
     }
 
     inner class PageChangeListener : ViewPager.OnPageChangeListener {
+
         override fun onPageScrollStateChanged(p0: Int) {}
 
         override fun onPageScrolled(position: Int, offsetPercentage: Float, offsetPixels: Int) {
-            pager.adapter?.count?.let { count ->
-                if (position == count - 2) {
-                    newsContainer.alpha = 1.0f - offsetPercentage
-                    val translation = -(screenWidth * offsetPercentage)
-                    proceed.translationX = translation
-                    topBar.translationX = translation
-                    pagerIndicator.translationX = translation
-                }
-                if (position == count - 1 && offsetPercentage == 0f) {
-                    onDismiss()
-                    dialog?.dismiss()
+            binding.apply {
+                pager.adapter?.count?.let { count ->
+                    when (items.last()) {
+                        is DismissiblePagerModel.SwipeOffScreen->{
+                            if (position == count - 2) {
```

Comment:
I think we can perhaps remove these position checks in our new paradigm with the `DismissiblePagerModel`. Let's investigate that together.

Reply:
Dont we need these to check if we are on the page next to the `SwipeOffScreen` fragment?